logicrunner/builtin/contract/migrationadmin/migrationadmin.go

```diff
@@ -0,0 +1,121 @@
+ */
+
+package migrationadmin
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/insolar/insolar/insolar"
+	"github.com/insolar/insolar/logicrunner/builtin/foundation"
+)
+
+// MigrationAdmin manage and change status for  migration daemon.
+type MigrationAdmin struct {
+	foundation.BaseContract
+	MigrationDaemon      foundation.StableMap
+	MigrationAdminMember insolar.Reference
+}
+
+const (
+	StatusActive     = "ACTIVE"
+	StatusInactivate = "INACTIVE"
+)
+
+// Create new Migration admin in genesis.
+func NewMigrationAdmin(migrationDaemons [insolar.GenesisAmountMigrationDaemonMembers]insolar.Reference, migrationAdminMember insolar.Reference) (*MigrationAdmin, error) {
+	daemonMigration := make(foundation.StableMap)
+	for i := 0; i < insolar.GenesisAmountMigrationDaemonMembers; i++ {
+		daemonMigration[migrationDaemons[i].String()] = StatusInactivate
+	}
+	return &MigrationAdmin{MigrationDaemon: daemonMigration, MigrationAdminMember: migrationAdminMember}, nil
+}
+
+// Return stable map migration daemon.
+func (mA MigrationAdmin) GetAllMigrationDaemon() (foundation.StableMap, error) {
+	sizeMap := len(mA.MigrationDaemon)
+	if sizeMap != insolar.GenesisAmountMigrationDaemonMembers {
+		return foundation.StableMap{}, fmt.Errorf(" MigrationAdmin contains the wrong amount migration daemon %d", sizeMap)
+	}
+	return mA.MigrationDaemon, nil
+}
+
+// Activate migration daemon.
+func (mA MigrationAdmin) ActivateDaemon(daemonMember string, caller insolar.Reference) error {
+	if caller == mA.MigrationAdminMember {
+		return fmt.Errorf(" only migration admin can activate migration demons ")
+	}
+	switch mA.MigrationDaemon[daemonMember] {
+	case StatusActive:
+		return fmt.Errorf(" daemon member already activated - %s", daemonMember)
+	case StatusInactivate:
+		mA.MigrationDaemon[daemonMember] = StatusActive
+		return nil
+	default:
+		return fmt.Errorf(" this referense is not daemon member ")
+	}
+}
+
+// Deactivate migration daemon.
+func (mA MigrationAdmin) DeactivateDaemon(daemonMember string, caller insolar.Reference) error {
+	if caller == mA.MigrationAdminMember {
+		return fmt.Errorf(" only migration admin can deactivate migration demons ")
+	}
+	switch mA.MigrationDaemon[daemonMember] {
+	case StatusActive:
+		mA.MigrationDaemon[daemonMember] = StatusInactivate
+		return nil
+	case StatusInactivate:
+		return fmt.Errorf(" daemon member already deactivated - %s", daemonMember)
+	default:
+		return fmt.Errorf(" this referense is not daemon member ")
+	}
+}
+
+// Check this member is migration daemon or mot.
+func (mA MigrationAdmin) CheckActiveDaemon(daemonMember string) (bool, error) {
+	status := mA.MigrationDaemon[daemonMember]
+	if status == StatusActive {
+		return true, nil
+	}
+	return false, fmt.Errorf(" this referense is not  active daemon member %s", daemonMember)
+}
+
+// Return Information about migration admin and migration daemons.
+func (mA MigrationAdmin) Info() ([]byte, error) {
+
+	res := map[string]interface{}{
```

Comment:
Use InfoResponse struct

Reply:
please look analog method in rootdomain

Reply:
This method in rootDomain is needed only for genesisdataprovider.

Reply:
get info from genesisrefs